src/main/java/duke/parsers/InputParser.java

```diff
@@ -0,0 +1,205 @@
+package duke.parsers;
+
+
+import duke.exceptions.DukeException;
+import duke.filehandler.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Scanner;
+
+public class InputParser {
+
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    /***
+     * Checks user inputs, if invalid throws DukeException
+     * @param str User input split by each word
+     * @param task type of task - todo,event,deadline,mark/unmark
+     * @throws DukeException
+     */
+    public static void inputChecker(String[] str, String task) throws DukeException {
+
+        if (str.length < 2) {
+            throw new DukeException("☹ OOPS!!! The description of a " + task + " cannot be empty.");
+        }
+    }
+
+    /***
+     * receives string of date and time, returns Date object
+     * @param str
+     * @return Date
+     * @throws DukeException
+     */
+    public static Date dateParser(String str) throws DukeException {
+        if (str.length() < 15) {
+            throw new DukeException("Invalid date, must be of the form dd/mm/yyyy hhmm");
+        }
+        String newStr = str.substring(0, 13) + ":" + str.substring(13);
+        System.out.println(newStr);
+        DateFormat df = new SimpleDateFormat("dd/MM/yyyy HH:mm");
+        try {
+            Date d1 = df.parse(newStr);
+            return d1;
+        } catch (Exception e) {
+            System.out.println(e);
+            throw new DukeException("Invalid date");
+        }
+    }
+
+    /***
+     * parses user input
+     * @param input
+     * @param toStore
+     * @return truw if user has not exited, false if user has exited chatbot
+     */
+
+    public static boolean parse(String input, boolean toStore) {
+        String[] splitStr = input.trim().split("\\s+");
+
+        if (input.equals("bye")) {
+            return false;
+        }
+        //list out each task from Duke.tasks ArrayList
+        else if (input.equals("list")) {
```

Comment:
For `if` blocks that you want to add comments on, perhaps consider moving the comment to within the code block? That way, your code follows Egyptian-style brackets.

```suggestion
        } else if (input.equals("list")) {
            //list out each task from Duke.tasks ArrayList
```

This applies to other similar sections as well.

Additionally, with this many cases, perhaps you might want to consider using `switch`-`case`? IDEs are likely to also be able to auto-convert them for you.